students/eowyn/session04/revised_mailroom.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+import sys
+
+''' global variables'''
+donor_names = ["Margaret Atwood", "Fred Armisen",
+               "Heinz the Baron Krauss von Espy"]
+donations = [[300, 555], [240, 422, 1000], [1500, 2300]]
+DONORS = dict(zip(donor_names, donations))
+
+
+def mainloop():
+    ''' main interactive loop 
+    "send a thank you" "create a report" or "quit"
+    '''
+    while True:
+        answer = input("Select from one of these options: \n"
+                       "(1) Send a Thank You\n"
+                       "(2) Create a Report\n"
+                       "(3) Send letters to everyone\n"
+                       "(4) Quit\n>")
+        arg_dict = {"1": thank_you,
+                    "2": print_report,
+                    "3": send_letters,
+                    "4": quit_code}
+        try:
+            arg_dict.get(answer)()
+        except TypeError:
```

Comment:
what's going to raise a TypeError?

You'll get a KeyError if they answer something that isn't in your dict.